### PR TITLE
GSP-118: Update RFC Format and Process

### LINKED ISSUE #110: Idea: update rfc format
The current proposal process and format is specified by https://github.com/beyondstorage/specs/blob/master/rfcs/17-proposal-process.md and https://github.com/beyondstorage/specs/blob/master/spec/2-proposal.md.

The metadata `status` and `updated_at` are required.

For `status`, the transition `draft` -> `candidate` -> `finished` is troublesome (have to start a PR to update status). In practice, this is not followed (There are 18 rfcs with status `draft`. Actually, we implemented a proposal immediately after an RFC. 
As an alternative, we can link an issue to track the implementation status.

For `updated_at`, we often forgot to update this when a proposal was discussed for days. And I think maybe the date information is not important (or only give `created_at`?

---
ref: Rust RFC https://github.com/rust-lang/rfcs

metadata
```markdown
- Feature Name: (fill me in with a unique ident, `my_awesome_feature`)
- Start Date: (fill me in with today's date, YYYY-MM-DD)
- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
```

### PR DESCRIPTION
**Pull request by xxchan**
Early discussion:
- closes #110

### CODE REVIEW

File: rfcs/0-example.md [hunk 1]
```diff
@@ -1,10 +1,15 @@
...
-# Proposal: <proposal name>
+- Author: (fill me in with `name <mail>`, e.g., Xuanwo <github@xuanwo.io>)
+- Start Date: (fill me in with today's date, YYYY-MM-DD)
+- RFC PR: [beyondstorage/specs#0](https://github.com/beyondstorage/specs/issues/0)
+- Tracking Issue: [beyondstorage/go-storage#0](https://github.com/beyondstorage/go-storage/issues/0)
+- Updates: (delete this part if not applicable)
+  - [GSP-20](./20-abc): Deletes something
+- Updated By: (delete this part if not applicable)
+  - [GSP-10](./10-do-be-do-be-do): Adds something
+  - [GSP-1000](./1000-lalala): Deprecates this RFC
```
Comment (xxchan) [inline rfcs/0-example.md:9]: Or put them in another section to keep the header clean?

File: README.md [hunk 2]
```diff
@@ -1,3 +1,11 @@
 # Specs
 
 This repo maintains rfcs and definitions for Beyond Storage.
+
+## RFC Process
+- (Optional) Submit an issue in `beyondstorage/specs` or [forum](https://forum.beyondstorage.io/) as a pre-proposal to discuss the idea.
```
Comment (Xuanwo) [inline README.md:6]: Leave an empty line after the header.

File: README.md [hunk 3]
```diff
@@ -1,3 +1,11 @@
...
+- Copy `rfcs/0-example.md` to `rfcs/0-my-feature.md` (where "my-feature" is descriptive). Don't assign an RFC number yet; This is going to be the PR number.
+- Fill in the RFC.
+- Submit a pull request. Use the issue number of the PR to update the filename.
```
Comment (Xuanwo) [inline README.md:9]: If the proposal is huge or complex, the reviewer could require the author to implement a demo.

File: rfcs/0-example.md (hunk 1 above)
Comment (Xuanwo) [reply rfcs/0-example.md:9]: I support to put them in another section.

**Review by Xuanwo — APPROVED** [review]

**Comment by Xuanwo** [comment]
Ping @JinnyYi for review.

**Review by JinnyYi — APPROVED** [review]

**Review by Xuanwo — APPROVED** [review]

**Pull request merged**